### PR TITLE
Build mgmt from a Fedora container

### PR DESCRIPTION
**Pull request by laurent-indermuehle**
This PR provides a way to build mgmt for Fedora using a container.
When following the [quick start guide](https://github.com/purpleidea/mgmt/blob/master/docs/quick-start-guide.md) for the first time, none of the offered methods to build mgmt worked.

To ensure regel 6.1 is installed instead of the 7.0, I had to refactor test/util.sh and misc/make-deps.sh. This is why the `in_ci` method became `in_env`.

Maybe it would have been better to just rebuild the RPM for latest version of Fedora. But I prefer to use a container to keep my workstation clean. Another idea would be to use github action. 

In the meantime this provide a solution, today, for newcomers.

### CODE REVIEW

File: docs/quick-start-guide.md [hunk 1]
```diff
@@ -21,6 +21,8 @@ to build your own.
 
 ### Downloading a pre-built release:
 
+This method is not recommanded because those packages are now very old (0.21)
```
Comment (purpleidea) [inline docs/quick-start-guide.md:24]: Thank you!
Small typo:

recommanded -> recommended

Bonus: Remove the ` (0.21)` and put a period at the end of the sentence instead.

File: docs/quick-start-guide.md [hunk 2]
```diff
@@ -95,6 +97,22 @@ At the moment we have:
 Please contribute more and help improve these! We'd especially like to see a
 Debian package!
 
+
+### Building from a container
+
+This method avoid polluting your workstation with the dependencies for the build. Here is an example using Fedora, Podman and Buildah :
```
Comment (purpleidea) [inline docs/quick-start-guide.md:103]: s/avoid/avoids/

(Petit `s` qui manque, svp)

File: test/util.sh [hunk 3]
```diff
@@ -20,9 +20,9 @@ fi
 
 TIMEOUT="$timeout --kill-after=360s --signal=QUIT 300s"
 
-in_ci() {
+in_env() {
```
Comment (purpleidea) [inline test/util.sh:23]: Looks great!

File: test/util.sh [hunk 4]
```diff
@@ -42,7 +44,7 @@ in_ci() {
 }
 
 fail_test() {
-	if in_ci github; then
+	if q github; then
```
Comment (purpleidea) [inline test/util.sh:47]: q?

File: docs/quick-start-guide.md [hunk 5]
```diff
@@ -95,6 +97,22 @@ At the moment we have:
 Please contribute more and help improve these! We'd especially like to see a
 Debian package!
 
+
+### Building from a container
```
Comment (purpleidea) [inline docs/quick-start-guide.md:101]: Add : if you like

**Review by purpleidea — COMMENTED** [review]
Thanks for the patches!

I made a few small recommendations for fixes. You can either put all the fixes and these three commits all together into a single commit, or rebase them into the three logical commits with fixes.

Cheers!

File: docs/quick-start-guide.md (hunk 5 above)
Comment (purpleidea) [inline docs/quick-start-guide.md:101]:
```suggestion
### Building from a container:
```

**Comment by purpleidea** [comment]
LGTM, let's see what CI says. Cheers!

**Comment by purpleidea** [comment]
> ./test/test-markdownlint.sh
FAILED

docs/quick-start-guide.md:100: MD012 Multiple consecutive blank lines
docs/quick-start-guide.md:115: MD012 Multiple consecutive blank lines
docs/quick-start-guide.md:103: MD013 Line length


Also please change:

`Buildah :`

to:

`Buildah:`

**Comment by purpleidea** [comment]
Merged, thank you very much!

Now that you've got your first patch in, lmk if you need help deciding on what to do for patch number two! Each one gets more fun.

Cheers! :trophy:

**Comment by laurent-indermuehle** [comment]
@purpleidea Thank you for your trust ! And thanks for the proposition for patch n°2, I'll keep in touch on Matrix.

**Pull request merged**